### ISSUE TITLE
Search for GC code fails

### ISSUE THREAD

**Issue by geocachermgo** [cgeo/cgeo#11740]
### Describe your problem!

Put GC1EDPO in seach window
No result and filter "cache name contains" is active

### How to reproduce?

Put GC1EDPO in seach window at start window

### Actual result after these steps?

No result and filter "cache name contains" is active

### Expected result after these steps?

GC1EDPO should be found

### Reproducible

Yes

### c:geo Version

2021.09.18-NB-3a7d785

**Comment by moving-bits** [comment]
If you are searching with a geocode ending on the letter "O", this would be an invalid geocode. (gc.com does a URL rewrite in this case!)
If in contrast your geocode end with the number "0", c:geo will indeed find a geocache and download its data.

Can it be that you have a name filter active, that does not match that cache's name, and that gets applied here?

**Comment by eddiemuc** [comment]
Indeed the main screen searches for given text as keyword / cache name and not for geocode. Whether this was different in former releases (And is thus a bug/regression) I cannot tell

**Comment by moving-bits** [comment]
@geocachermgo 
Which search function are you using in this case? Mainscreen's search in the actionbar (magnifier icon), or any of the searches offered by tapping on the "search" icon on mainscreen?
If the latter, please retry with the "geocode" search - that has worked for me with your geocode (but ending on the number "0", not the letter "O").

**Comment by Lineflyer** [comment]
The main screen search also searches for geocodes and works well, but GC1EDPO is no valid geococde. I don't understand the issue.

**Comment by geocachermgo** [comment]
I searched with the mainscreen action bar, that finds other GC codes.
I copied the code from an app whith a font were 0 and O are the same.
The link https://coord.info/GC1EDPO leads directy to GC1EDP0 which I took wrongly as a confirmation that the GC code is a correct one.
I tried https://coord.info/GC1EDOB which leads to GC1ED0B. Analogous for GCC0DE. Seems like possible GC codes with O are replaced by those with 0. If this is a general principle at GC we could think about extending the search to that.
